app/models/auth_token.rb

```diff
@@ -15,6 +15,8 @@
 class AuthToken < ApplicationRecord
   validates_presence_of :token
   scope :authorized, -> { where(authorized: [true, nil]) }
+  # find tokens that include ANY of the scopes provided
+  scope :with_either_scope, ->(required_scope) { where("scopes && array[?]::varchar[]", Array(required_scope)) }
```

Comment:
The name of this scope is a little weird. I wanted something to imply that if any of the scope names match then a result will be returned rather than having to match all of the scopes passed in.

Reply:
`has_scope`?